src/media-server.js

```diff
@@ -0,0 +1,148 @@
+import { once } from 'events'
+import fastify from 'fastify'
+import pTimeout from 'p-timeout'
+import StateMachine from 'start-stop-state-machine'
+
+import BlobServerPlugin from './fastify-plugins/blobs.js'
+import { kBlobStore } from './mapeo-project.js'
+
+export const BLOBS_PREFIX = 'blobs'
+export const ICONS_PREFIX = 'icons'
+
+/**
+ * @typedef {Object} StartOpts
+ *
+ * @property {string} [host]
+ * @property {number} [port]
+ */
+
+export class MediaServer {
+  #serverState
+  #fastify
+  #createFastify
+
+  /**
+   * @param {object} params
+   * @param {(projectPublicId: string) => Promise<import('./mapeo-project.js').MapeoProject>} params.getProject
+   * @param {import('fastify').FastifyServerOptions['logger']} [params.logger]
+   */
+  constructor({ getProject, logger }) {
+    this.#createFastify = () => {
+      const server = fastify({ logger })
+
+      server.register(BlobServerPlugin, {
+        prefix: BLOBS_PREFIX,
+        getBlobStore: async (projectPublicId) => {
+          const project = await getProject(projectPublicId)
+          return project[kBlobStore]
+        },
+      })
+
+      return server
+    }
+
+    this.#fastify = this.#createFastify()
+
+    this.#serverState = new StateMachine({
+      start: this.#startServer.bind(this),
+      stop: this.#stopServer.bind(this),
+    })
+  }
+
+  /**
+   * @param {StartOpts} [opts]
+   */
+  async #startServer({ host = '127.0.0.1', port } = {}) {
```

Comment:
does it make sense to bind to `127.0.0.1` by default?

Reply:
Yes that makes sense. This should mean that the server cannot be accessed from another machine on the same network, which is what we want. In the future we can add a debug option to listen on `0.0.0.0` so that we can test from a laptop on the same wifi.